call-home.sh

```diff
@@ -0,0 +1,286 @@
+  -i  [PERCONA_INSTANCE_ID]             Instance id                                                 [Default: autogenerated]
+  -j  [PERCONA_INSTANCE_ID_FILE_PATH]   Path of the file where to store the unique ID of this node. [Default: $PERCONA_INSTANCE_ID_FILE_PATH]
+  -u  [PERCONA_TELEMETRY_URL]           Percona Telemetry Service endpoint                          [Default: $PERCONA_TELEMETRY_URL]
+  -t  [PERCONA_SEND_TIMEOUT]            Default timeout for the curl command.                       [Default: $PERCONA_SEND_TIMEOUT]
+
+For example,
+on a CentOS7, you may run the script as:
+
+./$0 -f "PRODUCT_FAMILY_MYSQL" -v "8.0.33" -s "\$(cat /etc/redhat-release)" -i "13f5fc62-35b4-4716-b3e6-96c761fc204d" -j /tmp/percona.telemetry -u "https://TO/DO/PROVIDE/ENDPOINT -t 1
+
+on Ubuntu, you may run the script as:
+./$0 -f "PRODUCT_FAMILY_MYSQL" -v "8.0.33" -s "\$(cat /etc/issue)" -i "13f5fc62-35b4-4716-b3e6-96c761fc204d" -j /tmp/percona.telemetry -u "https://TO/DO/PROVIDE/ENDPOINT -t 1
+
+EOF
+
+    if [[ $errorCode -ne 0 ]];
+    then
+        exit_script $errorCode
+    fi
+}
+
+# Perform any required cleanup and exit with the given error/success code
+exit_script()
+{
+    # Exit with a given return code or 0 if none are provided.
+    exit ${1:-0}
+}
+
+# Vaildate arguments to ensure that mandatory ones have been provided
+validate_args()
+{
+    local USAGE_TEXT="See usage for details."
+
+    if [[ -z "$PERCONA_PRODUCT_FAMILY" ]];
+    then
+        printf "PERCONA_PRODUCT_FAMILY not provided. %s\n" "$USAGE_TEXT" >&2
+        usage 1
+    fi
+
+    if [[ -z "$PERCONA_PRODUCT_VERSION" ]];
+    then
+        printf "PERCONA_PRODUCT_VERSION not provided. %s\n" "$USAGE_TEXT" >&2
+        usage 1
+    fi
+
+    if [[ -z "$PERCONA_OPERATING_SYSTEM" ]];
+    then
+        printf "PERCONA_OPERATING_SYSTEM not provided. %s\n" "$USAGE_TEXT" >&2
+        usage 1
+    fi
+
+    if [[ -z "$PERCONA_DEPLOYMENT_METHOD" ]];
+    then
+        printf "PERCONA_DEPLOYMENT_METHOD not provided. %s\n" "$USAGE_TEXT" >&2
+        usage 1
+    fi
+
+    if [[ -z "$PERCONA_INSTANCE_ID" ]];
+    then
+        printf "PERCONA_INSTANCE_ID not provided. %s\n" "$USAGE_TEXT" >&2
+        usage 1
+    fi
+}
+
+# Creates the instance id file if one doesn't exist
+# If instance id file exists, we already reported this instance and there is no more work to do
+check_or_create_percona_instance_id_file()
+{
+    if [[ -f "$PERCONA_INSTANCE_ID_FILE_PATH" ]];
+    then
+        exit_script
+    fi
+    
+    # If instance ID is not provided externally, generate it
+    if [[ -z "$PERCONA_INSTANCE_ID" ]];
+    then
+        PERCONA_INSTANCE_ID=$(cat /proc/sys/kernel/random/uuid)
+    fi
+
+    echo $PERCONA_INSTANCE_ID > $PERCONA_INSTANCE_ID_FILE_PATH
+}
+
+collect_data_for_report()
+{
+    json_message_map["id"]=$(cat /proc/sys/kernel/random/uuid)
+    json_message_map["createTime"]="$(date +"%Y-%m-%dT%H:%M:%S%:z")"
```

Comment:
actually we can't guarantee that user's time zone is UTC.

> date +"%Y-%m-%dT%H:%M:%S%:z"
> 2023-09-19T07:36:27+00:00

> date --iso-8601='seconds'
> 2023-09-19T07:36:37+03:00

Reply:
Actually, we can change the date command to force output in UTC timezone.

date -u '+%Y-%m-%d %H:%M %Z'

Reply:
```
$ date +"%Y-%m-%dT%H:%M:%S%:z"
2023-09-19T13:45:54+02:00
$ date --iso-8601='seconds'
2023-09-19T13:46:06+02:00

$ date -u '+%Y-%m-%d %H:%M %Z'
2023-09-19 11:46 UTC
```
So 1st and 2nd ones provide the same output. Should we use `date --iso-8601='seconds'`

@EngineeredVirus , 3rd option: Service API will reject it

Reply:
ok, up to implementor